### ISSUE TITLE
Expected curly rule schema in IDE different from what is documented

### ISSUE THREAD

**Issue by omganunicorn** [palantir/tslint#2961]
### Bug Report

- 5.4.3
- 2.3.4
- Webstorm 2017.1.4

#### TypeScript code being linted

The `tslint.json` configuration itself is where the warning happens:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
        "array-type": [true, "generic"],
        "ban-types": [true, ["Object", "Use {} instead."]],
        "curly": [true, "ignore-same-line"],
        "interface-name": [true, "never-prefix"],
        "no-namespace": [true, "allow-declarations"],
        "no-shadowed-variable": false,
        "quotemark": [true, "single", "jsx-single"]

    },
    "rulesDirectory": []
}
```

#### Actual behavior
`"curly": [true, "ignore-same-line"]` works when linting and single-line if statements do not show an error, however, Webstorm shows the "Type is not allowed" warning for the setting when editing `tslint.json`.

#### Expected behavior
The curly rule should accept the given data type correctly like the one documented in https://palantir.github.io/tslint/rules/curly/ for its options.

**Comment by ajafff** [comment]
There's nothing tslint can do to fix this.
This is an issue with Webstorm and Webstorm *probably* uses the schema from http://json.schemastore.org/tslint which is maintained externally.
If you really need it, you can contribute that change to https://github.com/schemastore/schemastore/

**Comment by omganunicorn** [comment]
So I looked into the issue a bit more this morning because I also thought it might be Webstorm (hence why I edited the title), but I was able to reproduce the issue in VS Code, with only [vscode-tslint](https://github.com/Microsoft/vscode-tslint) enabled. In Webstorm I am using the tslint installation from the node_modules folder and not the global or IDE built-in one.

I was going to close this as a non-issue since the error is being given when linting a JSON and not a .ts file, however I also looked at the node_modules/tslint/rules folder, and noticed that even thought the rule works (I see no problem with `/rules/curlyRule.d.ts` or `/rules/curlyRule.js`), curly _is_ a boolean in `recommended.d.ts`:

```
export declare const jsRules: {
    "align": {
        options: string[];
    };
    "class-name": boolean;
    "curly": boolean;
...
}
```

And in recommended.js notice how align has the proper strings `option` array but curly is a boolean, again, contrary to the documentation:

```
exports.jsRules = {
    "align": {
        options: [
            "parameters",
            "statements",
        ],
    },
    "class-name": true,
    "curly": true,
...
}
```

So even though the issue started in a JSON file, I think this needs to be fixed to avoid any other possible problems in the future; I'll look into the fix and make a PR later on.

**Comment by omganunicorn** [comment]
Closing for expected behavior and looking into the issue on the IDE side.

**Comment by BehindTheMath** [comment]
Looks like this was fixed with the following PR: https://github.com/SchemaStore/schemastore/pull/318

**Comment by ileathan** [comment]
Im still getting this bug, and also finding it happens for multiple rules.

How exactly did you fix it?

**Comment by BehindTheMath** [comment]
Let me clarify.

The schema checking in IDEs are not managed by this project. I believe VS Code uses the [SchemaStore's schema](https://github.com/SchemaStore/schemastore/blob/0f3dafbe8870e6cecfd7b39b01eefb963eb2e96c/src/schemas/json/tslint.json), and IntelliJ uses their own [schema](https://github.com/JetBrains/intellij-plugins/blob/a04b66bc75059782daa43796eec35c7004e74496/tslint/resources/tslintJsonSchema/tslint-schema.json) (which is probably based on SchemaStore's).

The issue is that both of those schemas are out of date and inaccurate. The commit I referenced above fixes this specific issue in the original schema, but the schema still needs to updated in VS Code and IntelliJ.

The reason you're still getting the correct behavior is because these schemas are just used for checking the structure of the tslint.json file. The actual linting is done by TSLint which is not managed by the IDE, so it should be up to date.
